concepts/Common/Security Considerations/20 HTML Encoding/10 Encode User Input.md

```diff
@@ -0,0 +1,178 @@
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+    export class AppComponent {
+
+        editorValue = '';
+
+        onFormSubmit (e) {
+            // ...
+            // Encode this.editorValue here with your favorite sanitizing tool before sending this value to the server 
+            // ...
+            
+            e.preventDefault();
+        }
+    }
+
+    <!-- tab: app.module.ts -->
+    import { BrowserModule } from '@angular/platform-browser';
+    import { NgModule } from '@angular/core';
+    import { AppComponent } from './app.component';
+
+    import { DxHtmlEditorModule, DxButtonModule } from 'devextreme-angular';
+
+    @NgModule({
+        declarations: [
+            AppComponent
+        ],
+        imports: [
+            BrowserModule,
+            DxHtmlEditorModule,
+            DxButtonModule
+        ],
+        providers: [ ],
+        bootstrap: [AppComponent]
+    })
+    export class AppModule { }
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <form
+            action="your-action"
+            @submit="handleSubmit">
+            <DxHtmlEditor ... 
+                v-model:value="editorValue"
+            />
+                
+            <DxButton
+                :use-submit-behavior="true"
+                text="Submit the Form"
+            />
+        </form>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxHtmlEditor } from 'devextreme-vue/html-editor';
+    import { DxButton } from 'devextreme-vue/button';
+
+    export default {
+        components: {
+            DxHtmlEditor, 
+            DxButton
+        },
+        data: {
+            editorValue: ''
+        },
+        methods: {
+            handleSubmit(e) {
+                // ...
+                // Encode this.editorValue here with your favorite sanitizing tool before sending this value to the server 
+                // ...
+
+                e.preventDefault();
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React, { useCallback, useState } from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { HtmlEditor } from 'devextreme-react/html-editor';
+    import { Button } from 'devextreme-react/button';
+
+    const App = () => {
+        const [editorValue, setEditorValue] = useState("");
+
+        const onFormSubmit = useCallback((e) => {
+            // ...
+            // Encode editorValue here with your favorite sanitizing tool before sending this value to the server 
+            // ...
+
+            e.preventDefault();
+        }, []);
+
+
+        return (
+            <form
+                action="your-action"
+                onSubmit={onFormSubmit}
+                <HtmlEditor ... 
+                    value={editorValue}
```

Comment:
I'm afraid we forgot `onValueChange`. Without it, `editorValue` won't ever be updated.

Reply:
I did not forget it, I excluded it on purpose =) Do we need to make this change in all versions in React approach?

Reply:
Sure